### ISSUE TITLE
Fact: Display campaigns on fact view page

### ISSUE THREAD

**Issue by aaronschachter** [DoSomethingArchive/legacy-website#316]
Depends on #315 

Once campaigns_by_view is built, add the views for the different campaign fields into the `buildContent()` output in the `FactEntityController` class, to provide campaigns team ability to view what campaigns use a given fact.  
- Be sure to wrap this code within a `if (module_exists('dosomething_campaign'))` conditional, as dosomething_campaign is not a dependency for dosomething_facts.

**Comment by deezone** [comment]
@aaronschachter A few questions:
1. Is " campaigns_by_view" built? I found the issue "Campaign View Modes for Pitch and Action" : https://github.com/DoSomething/dosomething/pull/236 that appears to be related - am on the right path? Is this yet to be built?
2. Can you please expand on "add the views for the different campaign fields". Are there fields to be added to the Fact entries or Campaign entries that connect the two (entity reference)?
3. Ideally, should the listing of related campaigns for the Fact view page be a link so the user can navigate to the campaign - additional / separate issue perhaps?
4. Should the listing of the campaigns be on the

**Comment by deezone** [comment]
Ha! Found the related issue: https://github.com/DoSomething/dosomething/issues/315 

Duh! Helps if I read the issue description.

**Comment by aaronschachter** [comment]
1. Yup that's what #315 is for.
2. The fields on a campaign that refernce facts are field_fact_problem, field_fact_solution, field_facts.  These 3 different fields relate to the different displays we need: where each field is a contextual argument to the view.
3. A fact already has a page, fact/[fid].  This issue is to add the views to the page. The entity controller's buildContent method is what determines what to display on this page.  We need to add three different elements to the content array using `views_embed_view` for each of the 3 display modes.  We can walk thru details in person to clarify.
